File: api/v1/schemas/board_diagnosis.py
# -*- coding: utf-8 -*-
"""
===================================
板块诊断 API 模型
===================================

本文件唯一职责：复用内部板块诊断契约，为 API 层提供稳定的响应模型导出。
"""

from src.schemas.board_diagnosis import BoardDefinitionSummary, BoardDiagnosisResult

__all__ = [
    "BoardDefinitionSummary",
    "BoardDiagnosisResult",
]

File: apps/dsa-web/src/api/boardDiagnosis.ts
import apiClient from './index';
import { toCamelCase } from './utils';
import type { BoardDefinitionSummary, BoardDiagnosisResult } from '../types/boardDiagnosis';

export const boardDiagnosisApi = {
  listBoards: async (): Promise<BoardDefinitionSummary[]> => {
    const response = await apiClient.get<Record<string, unknown>[] | Record<string, unknown>>(
      '/api/v1/board-diagnosis/boards',
    );
    return toCamelCase<BoardDefinitionSummary[]>(response.data);
  },

  getDiagnosis: async (boardKey: string): Promise<BoardDiagnosisResult> => {
    const response = await apiClient.get<Record<string, unknown>>(`/api/v1/board-diagnosis/${boardKey}`);
    return toCamelCase<BoardDiagnosisResult>(response.data);
  },
};
